### PR TITLE
MRG: add module to test installed scripts

### PR DESCRIPTION
**Pull request by matthew-brett**
Module tests whether we can run f2py and return correct version.

Skip these tests if we are running in-place.

Tests fix in #5815

This PR passes all tests with #5815 merged.

### CODE REVIEW

File: numpy/tests/test_scripts.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+from __future__ import division, print_function, absolute_import
+
+import os
+from os.path import join as pjoin, isfile, dirname, basename
+import sys
+from subprocess import Popen, PIPE
+
+import numpy as np
+from numpy.compat.py3k import basestring, asbytes
+
```
Comment (charris) [inline numpy/tests/test_scripts.py]: Don't need all the blank lines.

**Comment by charris** [comment]
You should work in a branch, not master. Also, could you squash the commits together into one `TST : ...`?

File: numpy/tests/test_scripts.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        # on Windows, and the arguments might be file paths with spaces.
+        # On Unix the list elements are each separate arguments.
+        cmd = ['"{0}"'.format(c) if ' ' in c else c for c in cmd]
+    proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = proc.communicate()
+    if proc.poll() == None:
+        proc.terminate()
+    if check_code and proc.returncode != 0:
+        raise RuntimeError(
+            """Command "{0}" failed with
```
Comment (charris) [inline numpy/tests/test_scripts.py]: There will be a lot of indentation white space in this message.

File: numpy/tests/test_scripts.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
```
Comment (charris) [inline numpy/tests/test_scripts.py]: I have mixed feelings about these editor lines...

File: numpy/tests/test_scripts.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+""" Test scripts
+
+Test that we can run installed scripts
+"""
+from __future__ import division, print_function, absolute_import
+
+import os
+from os.path import join as pjoin, isfile, dirname, basename
```
Comment (charris) [inline numpy/tests/test_scripts.py]: Maybe `pathjoin`?

File: numpy/tests/test_scripts.py (hunk 1 above)
Comment (matthew-brett) [inline numpy/tests/test_scripts.py]: I put the blank lines in to break up import blocks into stdlib, common third-party, and testing.  But happy to take them out if you prefer.

**Comment by matthew-brett** [comment]
Sorry about the 'master' branch - I must have been tired when I did this push.   I decided to keep going on the master branch to keep this PR going, but I can close this, open another if you prefer.

File: numpy/tests/test_scripts.py (hunk 1 above)
Comment (charris) [inline numpy/tests/test_scripts.py]: Understand the temptation, but I think at some point the blank lines become distracting. Separating system and user modules is good, but after that careful ordering is usually sufficient. There are no rules for blank lines, so it's a judgement call.

File: .travis.yml [hunk 5]
```diff
@@ -28,6 +28,9 @@ before_install:
   - ulimit -a
   - mkdir builds
   - pushd builds
+  # Build into own virtualenv
```
Comment (charris) [inline .travis.yml]: Do we always want to do this? Just curious. A bit more explanation in the comment would help.

Comment (matthew-brett) [inline .travis.yml]: I think we do - I added a comment.

File: tools/travis-test.sh [hunk 6]
```diff
@@ -120,10 +120,17 @@ fi
...
   $PYTHON setup.py bdist_wheel
-  $PIP install --pre --upgrade --find-links dist numpy
+  # Make another virtualenv to install into
+  virtualenv --python=python venv-for-wheel
+  . venv-for-wheel/bin/activate
+  # Move out of source directory to avoid finding local numpy
+  cd dist
+  $PIP install --pre --upgrade --find-links . numpy
+  $PIP install nose
+  cd ..
```
Comment (argriffing) [inline tools/travis-test.sh]: The reason to not use `pushd` / `popd` is for portability across shells?

Comment (matthew-brett) [inline tools/travis-test.sh]: No good reason - it's not my personal habit - but I've switched it now.

File: numpy/tests/test_scripts.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
+""" Test scripts
+
+Test that we can run installed scripts
```
Comment (charris) [inline numpy/tests/test_scripts.py]: "installed" or "install"? Bit more explanation if the first.

**Comment by matthew-brett** [comment]
I rebased on master, so all the tests should pass.

**Comment by charris** [comment]
Travis is taking a nap...

**Comment by matthew-brett** [comment]
Oops - sh on Linux does not have pushd, in fact.

**Comment by charris** [comment]
? Mine does.
```
sh-4.3$ pushd doc
~/Workspace/numpy.git/doc ~/Workspace/numpy.git
sh-4.3$ popd
~/Workspace/numpy.git
```

**Comment by matthew-brett** [comment]
On my Mac, sh is aliased to bash - that was how I didn't notice first time.   Is it possible that is true on your system?

**Comment by njsmith** [comment]
Which shell you have at /bin/sh varies depending on distro and on local
configuration. So the advice is to either run /bin/bash if you want fancier
features that aren't guaranteed by POSIX, or else be careful to target the
lowest common denominator.

**Comment by charris** [comment]
Looks like mine is aliased also.
```
charris@localhost [numpy.git (master)]$ ll `which sh`
lrwxrwxrwx. 1 root root 4 Dec 30 12:30 /usr/bin/sh -> bash
```

**Comment by charris** [comment]
Thanks Matthew.

**Pull request merged**